workflow/controller/operator.go

```diff
@@ -269,16 +292,13 @@ func (woc *wfOperationCtx) operate() {
 		return
 	}
 
-	// Create a starting template context.
-	tmplCtx, err := woc.createTemplateContext(wfv1.ResourceScopeLocal, "")
-	if err != nil {
-		msg := fmt.Sprintf("Failed to create a template context: %+v", err)
-		woc.log.Errorf(msg)
-		woc.markWorkflowError(err, true)
-		return
-	}
+	var node *wfv1.NodeStatus
 
-	node, err := woc.executeTemplate(woc.wf.ObjectMeta.Name, &wfv1.WorkflowStep{Template: woc.wf.Spec.Entrypoint}, tmplCtx, woc.wf.Spec.Arguments, &executeTemplateOpts{})
+	if woc.wf.Spec.WorkflowTemplateRef != nil {
+		node, err = woc.executeTemplate(woc.wf.ObjectMeta.Name, &wfv1.WorkflowStep{TemplateRef: woc.convertWFTmplRefToTmplRef()}, tmplCtx, woc.arguments, &executeTemplateOpts{})
+	} else {
+		node, err = woc.executeTemplate(woc.wf.ObjectMeta.Name, &wfv1.WorkflowStep{Template: woc.wfSpec.Entrypoint}, tmplCtx, woc.arguments, &executeTemplateOpts{})
+	}
```

Comment:
This `if` statement is hard to read and is largely unnecessary. Why not something like:

```go
	tmpl := &wfv1.WorkflowStep{Template: woc.wfSpec.Entrypoint}
	if woc.wf.Spec.WorkflowTemplateRef != nil {
		tmpl.Template = ""
		tmpl.TemplateRef = woc.convertWFTmplRefToTmplRef()
	}
	node, err := woc.executeTemplate(woc.wf.ObjectMeta.Name, tmpl, tmplCtx, woc.wf.Spec.Arguments, &executeTemplateOpts{})
```
It's way cleaner...